internal/controller/backup_controller.go

```diff
@@ -570,6 +578,44 @@ func updateClusterWithSnapshotsBackupTimes(
 	return nil
 }
 
+// updateClusterWithPluginBackupTimes updates a Cluster's FirstRecoverabilityPoint
+// and LastSuccessfulBackup fields based on a completed plugin-based backup.
+func updateClusterWithPluginBackupTimes(
+	ctx context.Context,
+	cli client.Client,
+	backup *apiv1.Backup,
+) error {
+	wrapErr := func(msg string, err error) error {
+		return fmt.Errorf("in updateClusterWithPluginBackupTimes, %s: %w", msg, err)
+	}
+	var cluster apiv1.Cluster
+	if err := cli.Get(ctx, client.ObjectKey{
+		Namespace: backup.Namespace,
+		Name:      backup.Spec.Cluster.Name,
+	}, &cluster); err != nil {
+		return fmt.Errorf("could not fetch cluster %s/%s: %w", backup.Namespace, backup.Spec.Cluster.Name, err)
+	}
+
+	origCluster := cluster.DeepCopy()
+
+	firstPoint := backup.Status.StartedAt.Time
+	lastBackup := backup.Status.StoppedAt.Time
```

Comment:
I don't think that backup started and stopped times can reliably indicate the first and last points.

We detected these values with barman by iterating through the barman catalog, if I recall correctly

Reply:
Just to clarify further, even if we manage to save the correct  firstPoint and lastBackup it won't be enough. 

Those backups could be still deleted by the underlying plugin without the operator knowing, making the reported data wrong